containers/init/ready/ready.go

```diff
@@ -216,20 +211,46 @@ func main() {
 		}
 	}
 
-	var clientset kubernetes.Interface
-	kubeConfigFile, ok := os.LookupEnv(KubeConfigEnv)
-	if ok {
-		clientset, err = kubehelpers.ConnectWithConfig(kubeConfigFile)
+	kubeConfigFile, _ := os.LookupEnv(KubeConfigEnv)
+	schemebuilder := runtime.NewSchemeBuilder(func(scheme *runtime.Scheme) error {
+		scheme.AddKnownTypes(grpcv1.GroupVersion,
+			&grpcv1.LoadTest{},
+			&grpcv1.LoadTestList{},
+		)
+		metav1.AddToGroupVersion(scheme, grpcv1.GroupVersion)
+		return nil
+	})
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		if err != rest.ErrNotInCluster {
+			log.Fatalf("failed to connect within cluster: %v", err)
+		}
 		if err != nil {
-			log.Fatalf("failed to read kubeconfig: %v", err)
+			log.Fatalf("could not find a home directory for user: %v", err)
```

Comment:
nit: is this line executed for sure? The line 225 and line 229 are checking the same condition, right?